sr_data_visualization/src/sr_data_visualization/sr_data_visualizer_gui.py

```diff
@@ -53,13 +51,21 @@ def __init__(self, context):
         super().__init__(context)
 
         self.context = context
+        self.init_ui()
 
-        self._hand_finder = HandFinder()
-        self._hand_parameters = self._hand_finder.get_hand_parameters()
-        self.joint_prefix = next(iter(list(self._hand_parameters.joint_prefix.values())))
-        self.hand_joints = self._hand_finder.get_hand_joints()
+    def _detect_hand_id_and_joints(self):
+        self.joint_prefix = None
+        self.hand_joints = None
 
-        self.init_ui()
+        try:
+            joint_states_msg = rospy.wait_for_message("/joint_states", JointState, timeout=1)
+            self.joint_prefix = ([prefix for prefix in joint_states_msg.name if 'h' in prefix][0]).split("_")[0] + "_"
```

Comment:
self.joint_prefix = ([prefix for prefix in joint_states_msg.name if 'h_' in prefix][0]).split("_")[0] + "_"